temporal-sdk/src/main/java/io/temporal/worker/tuning/JVMSystemResourceInfo.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker.tuning;
+
+import com.sun.management.OperatingSystemMXBean;
+import io.temporal.common.Experimental;
+import java.lang.management.ManagementFactory;
+
+/** {@link SystemResourceInfo} implementation that uses JVM-specific APIs to get resource usage. */
+@Experimental
+public class JVMSystemResourceInfo implements SystemResourceInfo {
```

Comment:
Why is the class public? Shouldn't it be in internal?

Reply:
I think it's kind of a nice option that people can, in the event that they have some specific requirements around resourcing, but don't want to have to reimplement the whole resource-based tuner.